src/Middleware/LoginSessionController.php

```diff
@@ -36,36 +33,51 @@ class LoginSessionController extends LeftAndMain
      * @param HTTPRequest $request
      * @return HTTPResponse
      */
-    public function removeLoginSession(HTTPRequest $request): HTTPResponse
+    public function remove(HTTPRequest $request): HTTPResponse
     {
-        // Ensure CSRF protection
-        if (!SecurityToken::inst()->checkRequest($request)) {
-            return $this->jsonResponse(
-                ['errors' => 'Request timed out, please try again'],
-                400
-            );
-        }
+        return $this->removeLoginSession($request);
+    }
 
-        $id = $request->param('ID');
-        $loginSession = LoginSession::get()->byID($id);
-        if (!$loginSession) {
-            return $this->jsonResponse(
-                ['errors' => 'Something went wrong.'],
-                400
-            );
-        }
+    private function removeLoginSession(HTTPRequest $request): HTTPResponse
+    {
+        $failureMessage = _t(__CLASS__ . '.REMOVE_FAILURE', 'Something went wrong.');
+        try {
+            // Ensure CSRF protection
+            if (!SecurityToken::inst()->checkRequest($request)) {
+                return $this->jsonResponse([
```

Comment:
This should have a 400 response code.

Reply:
I've created an [new issue](https://github.com/silverstripe/silverstripe-session-manager/issues/45) to cover the status codes at the same time as writing unit tests

Reply:
I'd just like to point out that we were returning HTTP error codes prior to this PR and that retuning `'success' => false,` is something that got introduced by this PR.

I find it weird that we need a separate issue to address bad architectural decision we haven't merged yet.

Reply:
Status codes weren't being used by JS prior to this pull-request, instead we were using request.success see https://github.com/silverstripe/silverstripe-session-manager/pull/25/files#diff-739dcd81d6830719ec2169dbb42d39a3c7d6382fc9685f613cb3580cebd1efdfR38

`success => false` was added because it was missing before.  This PR did not add a new way of doing things.

From memory the non 200 status codes were causing some undesirable side effect (can't remember exactly what) so I just changed them all to 200 to resolve.

Reply:
The status codes weren't being used by the UI prior to this pull-request.  We were still relying on response.success - see https://github.com/silverstripe/silverstripe-session-manager/pull/25/files#diff-739dcd81d6830719ec2169dbb42d39a3c7d6382fc9685f613cb3580cebd1efdfR38

I added in `'success' => 'false'` because it was previously missing. This pull-request didn't change how things work.

The non-200 response codes were causing an unintentional side-effect (I cannot remember exactly what) so I changed them to 200 to resolve

If we want to refactor things from response.success to http response codes, then we can do so on the separate issue.